### PR TITLE
Stacks API version in response header

### PR DESCRIPTION
**Pull request by MHassanTariq**
## Description

This PR adds stacks API version in the response header as, 
`X-Stacks-API-Version : stacks-blockchain-api v3.0.3 (master:cd0c8aef)`

For details refer to issue #1186 

## Type of Change
- [X] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Checklist
- [ ] Code is commented where needed
- [ ] Unit test coverage for new or modified code paths
- [ ] `npm run test` passes
- [ ] Changelog is updated
- [ ] Tag 1 of @rafaelcr or @zone117x for review

### CODE REVIEW

File: src/api/init.ts [hunk 1]
```diff
@@ -128,6 +128,16 @@ export async function startApiServer(opts: {
     });
     app.use(promMiddleware);
   }
+  // adding stacks-api-version in header
+  app.use((_, res, next) => {
+    try {
+      const [branch, commit, tag] = fs.readFileSync('.git-info', 'utf-8').split('\n');
```
Comment (zone117x) [inline src/api/init.ts:134]: Should definitely avoid reading this file on _every_ request. These variables can be read and stored in a variable outside of this function.

File: src/api/init.ts [hunk 2]
```diff
@@ -128,6 +128,16 @@ export async function startApiServer(opts: {
...
+      res.setHeader('X-Stacks-API-Version', `stacks-blockchain-api ${tag} (${branch}:${commit})`);
```
Comment (zone117x) [inline src/api/init.ts:135]: This header needs added to CORs `Access-Control-Expose-Headers` allowlist if we want web browser js to be able to access it.

File: src/api/init.ts [hunk 3]
```diff
@@ -128,6 +137,14 @@ export async function startApiServer(opts: {
     });
     app.use(promMiddleware);
   }
+  // adding stacks-api-version in header
+  app.use((_, res, next) => {
+    if (stacksVersion) {
+      res.setHeader('X-Stacks-API-Version', stacksVersion);
```
[the remark this answers is not in the document]
Comment (CharlieC3) [reply src/api/init.ts:143]: Internally, we're restricting the content added to the `server` header for additional security. If we apply the version to that header, I'm not sure if it'll make it back to the origin, or if it'll be filtered out. We'd have to test it to be sure.

[the remark this answers is not in the document]
Comment (rafaelcr) [reply src/api/init.ts:143]: Ok thanks, no problem... in that case, using `X-Stacks-API-Version` sounds good to me

File: src/api/init.ts [hunk 4]
```diff
@@ -128,6 +137,14 @@ export async function startApiServer(opts: {
...
+    }
+    res.setHeader('Access-Control-Expose-Headers', 'X-Stacks-API-Version');
```
Comment (rafaelcr) [inline src/api/init.ts:145]: @MHassanTariq this would overwrite previous values for this header. Can you modify it so that it appends this new value?

**Review by zone117x — APPROVED** [review]
LGTM!

**Pull request merged**